pyomo/core/base/expression.py

```diff
@@ -170,34 +173,25 @@ def create_node_with_local_data(self, values):
         """
         obj = ScalarExpression()
         obj.construct()
-        obj.expr = values[0]
+        obj._args_ = values
         return obj
 
     #
     # Abstract Interface
     #
 
-    @property
-    def expr(self):
-        """Return expression on this expression."""
-        return self._expr
-
-    @expr.setter
-    def expr(self, expr):
-        self.set_value(expr)
-
     def set_value(self, expr):
         """Set the expression on this expression."""
         if expr is None or expr.__class__ in native_numeric_types:
-            self._expr = expr
+            self._args_ = (expr,)
```

Comment:
I assume it's intentional to make these tuples?

Reply:
Yes.  By convention `._args_` should always hold an iterable.  This PR is removing the "one-off" handling of named expressions (using an unnecessary `_expr` attribute).